### PR TITLE
Fix deadlocks on failures with parTraverse and parSequence

### PR DESCRIPTION
**Pull request by ngbinh**
Both of these tests fail by timing out on my local machine:

```scala
[info] BlazeClientSpec                                                                                     
[info] Blaze Http1Client should                                     
[info]   + raise error NoConnectionAllowedException if no connections are permitted for key (789 ms)
[info]   + make simple https requests (966 ms)                                                                                                      
[info]   + behave and not deadlock (1 second, 34 ms)                                         
[error]   x behave and not deadlock on failures with parTraverse (30 seconds, 245 ms)                      
[error]    None is not Some (BlazeClientSpec.scala:151)                               
[error] org.http4s.client.blaze.BlazeClientSpec.$anonfun$new$18(BlazeClientSpec.scala:151)
[error]   x behave and not deadlock on failures with parSequence (30 seconds, 308 ms)                              
[error]    None is not Some (BlazeClientSpec.scala:184)                                                                                  
[error] org.http4s.client.blaze.BlazeClientSpec.$anonfun$new$31(BlazeClientSpec.scala:184)                                                                                                                  
[info]   + obey response header timeout (30 seconds, 374 ms)                                                                                                                                                
[info]   + unblock waiting connections (30 seconds, 417 ms)                                                                                                       
[info]   + reset request timeout (30 seconds, 462 ms)                  
[info]   + drain waiting connections after shutdown (1 second, 140 ms)                                                                   
[info]   + cancel infinite request on completion (1 second, 193 ms)                                                                                                                                         
[info] Total for specification BlazeClientSpec                  
[info] Finished in 31 seconds, 904 ms                    
[info] 10 examples, 38 expectations, 2 failures, 0 error  
```

### CODE REVIEW

**Comment by ngbinh** [comment]
@rossabaker should this be reopened?

**Comment by rossabaker** [comment]
Yes, I mentioned it in the other ticket and it autoclosed.

**Comment by rossabaker** [comment]
After 212d7fa, the `parTraverse` test still fails for me due to #2265.  This fixes the deadlocks.

File: blaze-client/src/test/scala/org/http4s/client/blaze/BlazeClientSpec.scala [hunk 1]
```diff
@@ -145,11 +145,6 @@ class BlazeClientSpec extends Http4sSpec {
             }
 
           val allRequests = for {
-            _ <- failedRequests.handleErrorWith(_ => IO.unit)
-            _ <- failedRequests.handleErrorWith(_ => IO.unit)
-            _ <- failedRequests.handleErrorWith(_ => IO.unit)
-            _ <- failedRequests.handleErrorWith(_ => IO.unit)
-            _ <- failedRequests.handleErrorWith(_ => IO.unit)
             _ <- failedRequests.handleErrorWith(_ => IO.unit)
```
Comment (ngbinh) [inline blaze-client/src/test/scala/org/http4s/client/blaze/BlazeClientSpec.scala]: I remember there are cases where we need to `flatmap` two failures to reproduce the error.

**Comment by rossabaker** [comment]
The last commit should also fix #2265.

**Comment by ngbinh** [comment]
looks great to me. Thanks!

**Pull request merged**